### PR TITLE
Integrate ioi2017's communication task type into cms' communication task type

### LINKED ISSUE #19: Add a task type that takes manager's output to checker
We should have another task type similar to the Communication task, but the manager's output goes to the output file and checked by the checker instead of becoming the verdict.

@mirjalali pointed out that IOI 2017 HTC implemented this task type, available in https://github.com/ioi-2017/cms/blob/master/cms/grading/tasktypes/Communication2017.py. Investigate whether we can patch their code.

### PR DESCRIPTION
**Pull request by aguss787**
Resolves #19 

image

### CODE REVIEW

**Comment by aguss787** [comment]
Since there is a new configuration, existing problem must be re-saved to update its config

File: cms/grading/tasktypes/Communication.py [hunk 1]
```diff
@@ -33,11 +33,10 @@
 from cms.grading.steps import compilation_step, evaluation_step_before_run, \
     evaluation_step_after_run, extract_outcome_and_text, \
     human_evaluation_message, merge_execution_stats, trusted_step
-from cms.grading.tasktypes import check_files_number
+from cms.grading.tasktypes import check_files_number, eval_output
 from . import TaskType, check_executables_number, check_manager_present, \
     create_sandbox, delete_sandbox, is_manager_for_compilation
 
-
```
Comment (fushar) [inline cms/grading/tasktypes/Communication.py:40]: Super duper minor nit: please revert this; I think this is CMS's style to have 2 newlines after imports

File: cms/grading/tasktypes/Communication.py [hunk 2]
```diff
@@ -372,12 +391,12 @@ def evaluate(self, job, file_cacher):
...
         # limit we adjust the result.
         if box_success_user and evaluation_success_user and \
-                stats_user["execution_time"] >= job.time_limit:
+            stats_user["execution_time"] >= job.time_limit:
             evaluation_success_user = False
             stats_user['exit_status'] = Sandbox.EXIT_TIMEOUT
 
         success = box_success_user \
-            and box_success_mgr and evaluation_success_mgr
+                  and box_success_mgr and evaluation_success_mgr
```
Comment (fushar) [inline cms/grading/tasktypes/Communication.py:399]: Revert this change?

File: cms/grading/tasktypes/Communication.py [hunk 3]
```diff
@@ -372,12 +391,12 @@ def evaluate(self, job, file_cacher):
         # sandbox can only check its own; if the sum is greater than the time
         # limit we adjust the result.
         if box_success_user and evaluation_success_user and \
-                stats_user["execution_time"] >= job.time_limit:
+            stats_user["execution_time"] >= job.time_limit:
```
Comment (fushar) [inline cms/grading/tasktypes/Communication.py:394]: Revert this change?

**Review by fushar — CHANGES_REQUESTED** [review]

**Comment by aguss787** [comment]
Ah right, duh, auto formatting nya intellij :''

**Review by fushar — APPROVED** [review]

**Pull request merged**